src/App.tsx

```diff
@@ -1,11 +1,20 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [foundMovie, setFoundMovie] = useState<Movie | null>(null);
+
+  useEffect(() => {
+    if (foundMovie && movies.every(
+      movie => foundMovie.imdbId !== movie.imdbId,
+    )) {
+      setMovies([...movies, foundMovie]);
+    }
+  }, [foundMovie]);
```

Comment:
Looks like redundant state and useEffect. lets improve a bit
```suggestion
  const onMovieAdd = (foundMovie: Movie) => {
    const isExists = movies.some(({imdbId}) => foundMovie.imdbId === imdbId)
    
    if (!isExists) {
      setMovies([...movies, foundMovie]);
    }
  }
```